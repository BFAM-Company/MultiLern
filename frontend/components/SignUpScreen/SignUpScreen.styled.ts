import { StyleSheet, Platform } from "react-native";
import { StylesVariables } from "../../utils/GLOBALS";

export const styles = StyleSheet.create({
  mainContainer:{
      width:'100%',
      height:'100%',
      backgroundColor: 'white',
  },
  image:{
      width:'100%',
      height:'100%',
      display:'flex',
      
      alignItems:'center',
      objectFit: 'cover',
      ...Platform.select({
          web: {
              justifyContent:'center',
          },
          default: {
              justifyContent:'flex-end',
          }
      }),
      
  },
  fadingHeader:{
      width:'100%',
      display:'flex',
      alignItems:'center',
      justifyContent: 'center',
  },
  AccountCreateContainer:{
      height:'60%',
      display:'flex',
      justifyContent:'space-around',
      ...Platform.select({
          web: {
              alignItems:'center',
              width: '75%'
          },
          default: {
              width:'100%',	
              justifyContent: 'flex-end',
              marginBottom:20,
          }
      }),
  }, 
  LogoContainer:{
      width:'100%',
      display:'flex',
      flexDirection:'row',
      justifyContent:'center',
      alignItems:'center',
  },
  LogoImage:{
      width:StylesVariables.LogoSize,
      height:StylesVariables.LogoSize,        
      borderRadius:5,
  },
  titleText:{
      fontSize:30,
      color:'rgb(33,33,43)',
      fontWeight:'600',
      margin:10
  },
  mainTextContainer:{
      width:'100%',
      display:'flex',
      alignItems:'center',
      marginBottom: 10
  },
  mainText:{
      width:'90%',
      fontSize:19,
      fontWeight:'700',
      color:'gray',
      textAlign:'center',
  },
  mainContentContainer:{
      ...Platform.select({
          web: {
              width: '50%'
          },
          default: {
              width: '100%'
          }
      }),
      display:'flex',
      height: '100%',
      justifyContent:'space-around',
      alignItems:'center',
      margin:10,
  },
  formContainer: {
    width: '100%', 
    display: 'flex',
    margin: 10, 
    justifyContent:'space-around',
    alignItems:'center',
    height: '55%'
  },
  ORTextContainer: {
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    margin: 20
  },
  buttonsContainer: {
    width: '100%',
    display: 'flex',
    justifyContent:'space-around',
    alignItems:'center',
    height: '40%',
    margin: 10
  },
  input: {
      backgroundColor: 'white',
      width:'80%',
      height: StylesVariables.InputSize,
      borderRadius:20,
      display:'flex',
      flexDirection:'row',
      justifyContent:'space-between',
      alignItems:'center',
      margin:20,
      paddingLeft:40,
      paddingRight:40,
      shadowColor: "#000",
      shadowOffset: {
              width: 0,
              height: 8,
      },
      shadowOpacity: 0.44,
      shadowRadius: 10.32,

      elevation: 16,
  }
})
