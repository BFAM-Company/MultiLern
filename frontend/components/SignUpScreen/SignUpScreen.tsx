import React, {useCallback, useEffect, useState} from 'react';
import { Animated, Dimensions, TextInput } from 'react-native';
import { View, Text, SafeAreaView, StyleSheet, ImageBackground, Image, TouchableOpacity } from 'react-native';
import { useFonts } from 'expo-font';
import LinearGradient from 'react-native-linear-gradient';

function HomeScreen(props: any) {

    const fadeAnimHeader = React.useRef(new Animated.Value(0)).current;
    const fadeAnimContainer = React.useRef(new Animated.Value(0)).current;
    const [login, setLogin] = useState<string | undefined>(undefined);
    const [password, setPassword] = useState<string | undefined>(undefined);

    const [fontsLoaded, fontError] = useFonts({
        'MPLUSRounded1c-ExtraBold': require('./../../assets/fonts/MPLUSRounded1c-ExtraBold.ttf'),
    });

    useEffect(() => {
        fadeIn(fadeAnimHeader, 3000);
        fadeIn(fadeAnimContainer, 5000);
      }, []);

    const fadeIn = (animVariable: Animated.Value, durationValue: number | undefined) => {
        Animated.timing(animVariable, {
        toValue: 1,
        duration: durationValue,
        useNativeDriver: true,
        }).start();
    };

    
      if (!fontsLoaded && !fontError) {
        return null;
      }
    return (
        <View style={styles.mainContainer}>
            <ImageBackground
                source= {require('./../../assets/gradientBackground.png')}
                style={styles.image}
                resizeMode='cover'
                blurRadius={40}
            >
                <Animated.View
                    style={[
                        styles.fadingHeader,
                        {
                            opacity: fadeAnimHeader,
                        },
                    ]}>
                    <Text style={styles.fadingText}>Zaloguj się</Text>
                </Animated.View>

                <Animated.View
                    style={[
                        styles.fadingContainer,
                        {
                            opacity: fadeAnimContainer,
                        },
                    ]}>
                    <TextInput 
                        style={styles.input}
                        onChangeText={setLogin}
                        value={login}
                        placeholder='username lub email'/>
                    <TextInput 
                        style={styles.input}
                        onChangeText={setPassword}
                        value={password}
                        secureTextEntry={true} 
                        placeholder='password'/>
                    <TouchableOpacity style={styles.logInButton}
                        onPress = {
                            ()=>{
                                
                            }
                        }
                    >
                        <LinearGradient start={{x: 0, y: 0}} end={{x: 1, y: 0}} colors={['#4c669f', '#3b5998', '#192f6a']}>
                        <Text style={null}>Kontynuj jako gość</Text>
                        <Image source={require('./../../assets/lock-alt.png')} style={[styles.iconImage, {tintColor: 'white'}]}/>
                        </LinearGradient>
                        
                    </TouchableOpacity>
                </Animated.View>
            </ImageBackground>
        </View>
    );
}

const styles = StyleSheet.create({
    mainContainer:{
        width:'100%',
        height:'100%',
        backgroundColor: 'white',
    },
    image:{
        width:'100%',
        height:'100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'flex-start',
    },
    fadingHeader:{
        width:'100%',
        display:'flex',
        alignItems:'center',
        justifyContent: 'center',
        marginTop:100,
    },
    fadingContainer:{
        width:'90%',
        minHeight:300,
        display: 'flex',
        alignItems:'center',
        justifyContent: 'space-around',
        backgroundColor: '#fff',
        borderRadius: 20,
        marginTop:50,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 8,
        },
        shadowOpacity: 0.44,
        shadowRadius: 10.32,

        elevation: 16,
    },
    logoImage:{
        width:50,
        height:50,
    },
    fadingText:{
        color:'rgb(33,33,43)',
        fontWeight:'900',
        fontSize:34,
        letterSpacing:0.6,
        //fontFamily:'MPLUSRounded1c-ExtraBold'

    },
    input:{
        width:'90%',
        padding:5,
        borderBottomWidth:1,
        borderColor:'lightgray',
        fontSize:18,
    },
    logInButton:{

    },
})


export default HomeScreen;