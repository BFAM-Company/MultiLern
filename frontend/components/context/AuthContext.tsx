import AsyncStorage from "@react-native-async-storage/async-storage";
import React, { createContext, useState, ReactNode, useContext } from "react";
import { UserDataContext } from "./UserContext";

interface AuthState {
  accessToken: string | null;
  refreshToken: string | null;
  authenticated: boolean | null;
}

export interface AuthContextType {
  authState: AuthState;
  getAccessToken: () => string | null;
  setAuthState: (state: AuthState) => void;
  logout: () => Promise<void>;
}

const initialAuthState: AuthState = {
  accessToken: null,
  refreshToken: null,
  authenticated: false,
};

const AuthContext = createContext<AuthContextType | null>(null);
const { Provider } = AuthContext;

const AuthProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [authState, setAuthState] = useState<AuthState>(initialAuthState);
  const userContext = useContext(UserDataContext)


  const logout = async () => {
    await AsyncStorage.clear()
    setAuthState({
      accessToken: null,
      refreshToken: null,
      authenticated: false,
    });
    userContext?.setUserData({
      nickname: '',
      email: 'null',
      avatar: null,
      isLogged: false
    })
  };

  const getAccessToken = () => {
    return authState.accessToken;
  };

  return (
    <Provider
      value={{
        authState,
        getAccessToken,
        setAuthState,
        logout,
      }}
    >
      {children}
    </Provider>
  );
};

export { AuthContext, AuthProvider };
