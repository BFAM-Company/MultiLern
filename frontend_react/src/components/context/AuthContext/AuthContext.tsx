import React, { createContext, useState, ReactNode, Dispatch, SetStateAction } from 'react';

interface AuthState {
  accessToken: string | null;
  refreshToken: string | null;
  authenticated: boolean | null;
  isLoggingByGuest?: boolean | null;
}

export interface AuthContextType {
  authState: AuthState;
  getAccessToken: () => string | null;
  setAuthState: Dispatch<SetStateAction<AuthState>>;
  logout: () => void;
}

const initialAuthState: AuthState = {
  accessToken: null,
  refreshToken: null,
  authenticated: false,
};

const AuthContext = createContext<AuthContextType | null>(null);

const AuthProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [authState, setAuthState] = useState<AuthState>(initialAuthState);

  const logout = () => {
    localStorage.clear();
    setAuthState(initialAuthState);
  };

  const getAccessToken = () => authState.accessToken;

  return (
    <AuthContext.Provider value={{ authState, getAccessToken, setAuthState, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export { AuthContext, AuthProvider };
