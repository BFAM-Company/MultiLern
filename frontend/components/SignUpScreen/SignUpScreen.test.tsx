import React from 'react';
import { render, fireEvent, waitFor, cleanup } from '@testing-library/react-native';
import SignUpScreen from './SignUpScreen';
import { AuthContext } from '../context/AuthContext/AuthContext';
import { AxiosContext } from '../context/AxiosProvider/AxiosProvider';
import axios, { AxiosInstance } from 'axios';
import { Alert } from 'react-native';
import { openAuthSessionAsync, dismissBrowser } from 'expo-web-browser';


jest.mock("@react-native-async-storage/async-storage", () => require("../../__mocks__/mock-async-storage"));

jest.mock('expo-web-browser', () => ({
  openAuthSessionAsync: jest.fn().mockResolvedValueOnce({ type: 'success' }),
  dismissBrowser: jest.fn(),
}));

afterEach(cleanup);

describe('<SignUpScreen />', () => {
  it('renders correctly', () => {
    const mockPageSwitcher = jest.fn();
    const { toJSON } = render(<SignUpScreen pageSwitcher={mockPageSwitcher} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should switch screens when buttons are pressed', () => {
    const mockPageSwitcher = jest.fn();
    const { getByText } = render(<SignUpScreen pageSwitcher={mockPageSwitcher} />);

    fireEvent.press(getByText('Apple ID'));
    fireEvent.press(getByText('Facebook account'));

    expect(mockPageSwitcher).toHaveBeenCalledTimes(2);
  });

  it('should register successfully', async () => {
    const mockPublicAxios: AxiosInstance = axios.create();
    mockPublicAxios.post = jest.fn().mockResolvedValueOnce({});

    const mockSetAuthState = jest.fn();
    const mockGetAccessToken = jest.fn(() => 'testAccessToken');
    const mockLogout = jest.fn();

    const mockAuthContext = {
        setAuthState: mockSetAuthState,
        getAccessToken: mockGetAccessToken,
        logout: mockLogout,
        authState: {
            accessToken: 'mockAccessToken',
            refreshToken: 'mockRefreshToken',
            authenticated: true,
            isLoggingByGuest: false,
        }
    };

    const { getByPlaceholderText, getByText } = render(
      <AuthContext.Provider value={mockAuthContext}>
            <AxiosContext.Provider value={{ authAxios: mockPublicAxios, publicAxios: mockPublicAxios}}>
          <SignUpScreen pageSwitcher={() => { }} />
        </AxiosContext.Provider>
      </AuthContext.Provider>
    );

    const usernameInput = getByPlaceholderText('Podaj nazwę użytkownika');
    const emailInput = getByPlaceholderText('Podaj email');
    const passwordInput = getByPlaceholderText('Podaj hasło ');
    const repeatedPasswordInput = getByPlaceholderText('Powtórz hasło ');
    const signUpButton = getByText('Zarejestruj się');

    fireEvent.changeText(usernameInput, 'testuser');
    fireEvent.changeText(emailInput, 'test@example.com');
    fireEvent.changeText(passwordInput, 'testpassword');
    fireEvent.changeText(repeatedPasswordInput, 'testpassword');
    fireEvent.press(signUpButton);

    await waitFor(() => {
        expect(mockPublicAxios.post).toHaveBeenCalledTimes(1)
        expect(mockPublicAxios.post).toHaveBeenCalledWith('/users/create', {
            nickname: 'testuser',
            password: 'testpassword',
            email: 'test@example.com',
            logginMethod: 'Multilern',
        }, 
        expect.objectContaining({
            headers: {
                'Content-Type': 'application/json',
            },
        }));
    });
  });

    it('should handle registration failure', async () => {
        const mockPublicAxios: AxiosInstance = axios.create();
        // Symulacja błędu rejestracji
        mockPublicAxios.post = jest.fn().mockRejectedValueOnce({response: { data: { message: 'Registration failed' } } });

        const { getByText, getByPlaceholderText} = render(
            <AxiosContext.Provider value={{ publicAxios: mockPublicAxios, authAxios: mockPublicAxios }}>
                <SignUpScreen pageSwitcher={() => { }} />
            </AxiosContext.Provider>
        );

        const usernameInput = getByPlaceholderText('Podaj nazwę użytkownika');
        const emailInput = getByPlaceholderText('Podaj email');
        const passwordInput = getByPlaceholderText('Podaj hasło ');
        const repeatedPasswordInput = getByPlaceholderText('Powtórz hasło ');
        const signUpButton = getByText('Zarejestruj się');

        fireEvent.changeText(usernameInput, 'testuser');
        fireEvent.changeText(emailInput, 'test@example.com');
        fireEvent.changeText(passwordInput, 'testpassword');
        fireEvent.changeText(repeatedPasswordInput, 'testpassword');
        fireEvent.press(signUpButton);

        await waitFor(() => {
            expect(mockPublicAxios.post).toHaveBeenCalledTimes(1);
        });
    });


    it('should open authentication session with Discord', async () => {
        const { getByText } = render(<SignUpScreen pageSwitcher={() => { }} />);

        const discordButton = getByText('Discord');
        fireEvent.press(discordButton);

        await waitFor(() => {
            expect(openAuthSessionAsync).toHaveBeenCalledTimes(1);
            expect(dismissBrowser).toHaveBeenCalled();
        });
    });

    it('should display error message for repeated password input when passwords do not match', async () => {
        const mockPublicAxios: AxiosInstance = axios.create();
        mockPublicAxios.post = jest.fn().mockResolvedValueOnce({});

        const mockSetAuthState = jest.fn();
        const mockGetAccessToken = jest.fn(() => 'testAccessToken');
        const mockLogout = jest.fn();

        const mockAuthContext = {
            setAuthState: mockSetAuthState,
            getAccessToken: mockGetAccessToken,
            logout: mockLogout,
            authState: {
                accessToken: 'mockAccessToken',
                refreshToken: 'mockRefreshToken',
                authenticated: true,
                isLoggingByGuest: false,
            }
        };

        const { getByPlaceholderText, getByText, rerender } = render(
        <AuthContext.Provider value={mockAuthContext}>
                <AxiosContext.Provider value={{ authAxios: mockPublicAxios, publicAxios: mockPublicAxios}}>
            <SignUpScreen pageSwitcher={() => { }} />
            </AxiosContext.Provider>
        </AuthContext.Provider>
        );

        const passwordInput = getByPlaceholderText('Podaj hasło ');
        const repeatedPasswordInput = getByPlaceholderText('Powtórz hasło ');
        const signUpButton = getByText('Zarejestruj się');

        fireEvent.changeText(passwordInput, 'password1');
        fireEvent.changeText(repeatedPasswordInput, 'password2');
        fireEvent.press(signUpButton);

        // Wait for the component to update
        await waitFor(() => {
        // Rerender the component
        rerender(
            <AuthContext.Provider value={mockAuthContext}>
                <AxiosContext.Provider value={{ authAxios: mockPublicAxios, publicAxios: mockPublicAxios}}>
                <SignUpScreen pageSwitcher={() => { }} />
            </AxiosContext.Provider>
            </AuthContext.Provider>
        );

        // Check if the error message appears
        expect(getByText('⚠ Hasła muszą być identyczne')).toBeTruthy();
        });
    });

    

});
