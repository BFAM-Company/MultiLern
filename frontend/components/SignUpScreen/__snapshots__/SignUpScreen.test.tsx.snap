// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpScreen /> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      [
        {
          "backgroundColor": "white",
          "height": "100%",
          "width": "100%",
        },
        {
          "flex": 1,
        },
      ],
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    alwaysBounceHorizontal={false}
    alwaysBounceVertical={false}
    bounces={false}
    contentContainerStyle={
      {
        "flexGrow": 1,
      }
    }
    overScrollMode="never"
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "height": "100%",
            "justifyContent": "flex-end",
            "objectFit": "cover",
            "width": "100%",
          }
        }
      >
        <Image
          blurRadius={40}
          resizeMode="cover"
          source={1}
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <View
          style={
            {
              "display": "flex",
              "height": "60%",
              "justifyContent": "flex-end",
              "marginBottom": 20,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
                "opacity": 0,
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "borderRadius": 5,
                    "height": 75,
                    "width": 75,
                  }
                }
              />
              <Text
                style={
                  {
                    "color": "rgb(33,33,43)",
                    "fontSize": 30,
                    "fontWeight": "600",
                    "margin": 10,
                  }
                }
              >
                MultiLern
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "marginBottom": 10,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "gray",
                    "fontSize": 19,
                    "fontWeight": "700",
                    "textAlign": "center",
                    "width": "90%",
                  }
                }
              >
                Dołącz do nas i podnieść swoją naukę na wyższy poziom
              </Text>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "height": "100%",
                  "justifyContent": "space-around",
                  "margin": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "60%",
                    "justifyContent": "space-around",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "width": "100%",
                    }
                  }
                >
                  <TextInput
                    autoCapitalize="none"
                    autoCorrect={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="Podaj nazwę użytkownika"
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "borderRadius": 20,
                        "display": "flex",
                        "elevation": 16,
                        "flexDirection": "row",
                        "height": 93.38000000000001,
                        "justifyContent": "space-between",
                        "margin": 20,
                        "paddingLeft": 40,
                        "paddingRight": 40,
                        "shadowColor": "#000",
                        "shadowOffset": {
                          "height": 8,
                          "width": 0,
                        },
                        "shadowOpacity": 0.44,
                        "shadowRadius": 10.32,
                        "width": "80%",
                      }
                    }
                    value=""
                  />
                </View>
                <TextInput
                  autoCapitalize="none"
                  autoCorrect={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Podaj email"
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "display": "flex",
                      "elevation": 16,
                      "flexDirection": "row",
                      "height": 93.38000000000001,
                      "justifyContent": "space-between",
                      "margin": 20,
                      "paddingLeft": 40,
                      "paddingRight": 40,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.44,
                      "shadowRadius": 10.32,
                      "width": "80%",
                    }
                  }
                  value=""
                />
                <TextInput
                  autoCapitalize="none"
                  autoCorrect={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Podaj hasło "
                  secureTextEntry={true}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "display": "flex",
                      "elevation": 16,
                      "flexDirection": "row",
                      "height": 93.38000000000001,
                      "justifyContent": "space-between",
                      "margin": 20,
                      "paddingLeft": 40,
                      "paddingRight": 40,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.44,
                      "shadowRadius": 10.32,
                      "width": "80%",
                    }
                  }
                  value=""
                />
                <TextInput
                  autoCapitalize="none"
                  autoCorrect={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Powtórz hasło "
                  secureTextEntry={true}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "display": "flex",
                      "elevation": 16,
                      "flexDirection": "row",
                      "height": 93.38000000000001,
                      "justifyContent": "space-between",
                      "margin": 20,
                      "paddingLeft": 40,
                      "paddingRight": 40,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.44,
                      "shadowRadius": 10.32,
                      "width": "80%",
                    }
                  }
                  value=""
                />
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "display": "flex",
                      "elevation": 16,
                      "flexDirection": "row",
                      "height": 80.03999999999999,
                      "justifyContent": "space-between",
                      "margin": 15,
                      "opacity": 1,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.44,
                      "shadowRadius": 10.32,
                      "width": "80%",
                    }
                  }
                >
                  <ViewManagerAdapter_ExpoLinearGradient
                    colors={
                      [
                        4280361259,
                        4279045399,
                      ]
                    }
                    style={
                      {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "display": "flex",
                        "flexDirection": "row",
                        "height": "100%",
                        "justifyContent": "space-between",
                        "paddingLeft": 40,
                        "paddingRight": 40,
                        "width": "100%",
                      }
                    }
                    testID="gradient"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "rgb(33,33,43)",
                            "fontSize": 17,
                            "fontWeight": "700",
                          },
                          {
                            "color": "white",
                            "textAlign": "left",
                            "width": "auto",
                          },
                        ]
                      }
                    >
                      Zarejestruj się
                    </Text>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "display": "flex",
                          "flexDirection": "row",
                          "justifyContent": "space-around",
                          "margin": 10,
                        }
                      }
                    >
                      <Image
                        source={1}
                        style={
                          [
                            {
                              "height": 45,
                              "margin": 5,
                              "width": 45,
                            },
                            {
                              "tintColor": "white",
                            },
                          ]
                        }
                        testID="account-service-icon"
                      />
                    </View>
                  </ViewManagerAdapter_ExpoLinearGradient>
                </View>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "margin": 20,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "gray",
                      "fontSize": 19,
                      "fontWeight": "700",
                      "textAlign": "center",
                      "width": "90%",
                    }
                  }
                >
                  ~ LUB UŻYJ ~
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "35%",
                    "justifyContent": "space-around",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "display": "flex",
                      "elevation": 16,
                      "flexDirection": "row",
                      "height": 80.03999999999999,
                      "justifyContent": "space-between",
                      "margin": 15,
                      "opacity": 1,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.44,
                      "shadowRadius": 10.32,
                      "width": "80%",
                    }
                  }
                >
                  <ViewManagerAdapter_ExpoLinearGradient
                    colors={
                      [
                        4294967295,
                        4294967295,
                      ]
                    }
                    style={
                      {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "display": "flex",
                        "flexDirection": "row",
                        "height": "100%",
                        "justifyContent": "space-between",
                        "paddingLeft": 40,
                        "paddingRight": 40,
                        "width": "100%",
                      }
                    }
                    testID="gradient"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "rgb(33,33,43)",
                            "fontSize": 17,
                            "fontWeight": "700",
                          },
                          {
                            "color": undefined,
                            "textAlign": "left",
                            "width": "auto",
                          },
                        ]
                      }
                    >
                      Discord
                    </Text>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "display": "flex",
                          "flexDirection": "row",
                          "justifyContent": "space-around",
                          "margin": 10,
                        }
                      }
                    >
                      <Image
                        source={1}
                        style={
                          [
                            {
                              "height": 45,
                              "margin": 5,
                              "width": 45,
                            },
                            {
                              "tintColor": undefined,
                            },
                          ]
                        }
                        testID="account-service-icon"
                      />
                    </View>
                  </ViewManagerAdapter_ExpoLinearGradient>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "display": "flex",
                      "elevation": 16,
                      "flexDirection": "row",
                      "height": 80.03999999999999,
                      "justifyContent": "space-between",
                      "margin": 15,
                      "opacity": 1,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.44,
                      "shadowRadius": 10.32,
                      "width": "80%",
                    }
                  }
                >
                  <ViewManagerAdapter_ExpoLinearGradient
                    colors={
                      [
                        4294967295,
                        4294967295,
                      ]
                    }
                    style={
                      {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "display": "flex",
                        "flexDirection": "row",
                        "height": "100%",
                        "justifyContent": "space-between",
                        "paddingLeft": 40,
                        "paddingRight": 40,
                        "width": "100%",
                      }
                    }
                    testID="gradient"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "rgb(33,33,43)",
                            "fontSize": 17,
                            "fontWeight": "700",
                          },
                          {
                            "color": undefined,
                            "textAlign": "left",
                            "width": "auto",
                          },
                        ]
                      }
                    >
                      Apple ID
                    </Text>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "display": "flex",
                          "flexDirection": "row",
                          "justifyContent": "space-around",
                          "margin": 10,
                        }
                      }
                    >
                      <Image
                        source={1}
                        style={
                          [
                            {
                              "height": 45,
                              "margin": 5,
                              "width": 45,
                            },
                            {
                              "tintColor": undefined,
                            },
                          ]
                        }
                        testID="account-service-icon"
                      />
                    </View>
                  </ViewManagerAdapter_ExpoLinearGradient>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "display": "flex",
                      "elevation": 16,
                      "flexDirection": "row",
                      "height": 80.03999999999999,
                      "justifyContent": "space-between",
                      "margin": 15,
                      "opacity": 1,
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 8,
                        "width": 0,
                      },
                      "shadowOpacity": 0.44,
                      "shadowRadius": 10.32,
                      "width": "80%",
                    }
                  }
                >
                  <ViewManagerAdapter_ExpoLinearGradient
                    colors={
                      [
                        4294967295,
                        4294967295,
                      ]
                    }
                    style={
                      {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "display": "flex",
                        "flexDirection": "row",
                        "height": "100%",
                        "justifyContent": "space-between",
                        "paddingLeft": 40,
                        "paddingRight": 40,
                        "width": "100%",
                      }
                    }
                    testID="gradient"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "rgb(33,33,43)",
                            "fontSize": 17,
                            "fontWeight": "700",
                          },
                          {
                            "color": undefined,
                            "textAlign": "left",
                            "width": "auto",
                          },
                        ]
                      }
                    >
                      Facebook account
                    </Text>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "display": "flex",
                          "flexDirection": "row",
                          "justifyContent": "space-around",
                          "margin": 10,
                        }
                      }
                    >
                      <Image
                        source={1}
                        style={
                          [
                            {
                              "height": 45,
                              "margin": 5,
                              "width": 45,
                            },
                            {
                              "tintColor": undefined,
                            },
                          ]
                        }
                        testID="account-service-icon"
                      />
                    </View>
                  </ViewManagerAdapter_ExpoLinearGradient>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
